python/lsst/ts/genericcamera/driver/alliedvisioncamera.py

```diff
@@ -232,6 +238,92 @@ def stop_live_view(self):
                 self.camera.set_pixel_format(self.normal_image_type)
         super().stop_live_view()
 
+    def _run_streaming(self) -> None:
+        """Keep streaming mode for the camera alive."""
+        with vimba.Vimba.get_instance() as v:
+            # Need to get camera again because this follows
+            # start_streaming_mode and the context manager drops out
+            # rendering self.camera inoperable
+            camera = v.get_camera_by_id(self.id)
+            # Seem to need a log statement here for things to work
+            # Otherwise the context manager fails
+            self.log.info("Running _run_streaming")
+            with camera:
+                camera.GevTimestampControlReset.run()
+                self._set_streaming_start_information()
+                camera.start_streaming(
+                    handler=self._get_streaming_frame,
+                    buffer_count=self.frame_buffer_size,
+                    allocation_mode=vimba.AllocationMode.AnnounceFrame,
+                )
+                self.log.info("Begin streaming keep alive loop")
+                while self.run_streaming_task:
+                    time.sleep(0.001)
+                else:
+                    self.log.info("End streaming keep alive loop")
+                    camera.stop_streaming()
+                    self.streaming_mode_stop = ts_utils.current_tai()
+
+    def start_streaming_mode(self, exp_time: float, static_data: dict) -> None:
+        """Start image streaming mode for the camera.
+
+        Parameters
+        ----------
+        exp_time : `float`
+            The exposure time in seconds.
+        static_data : `dict`
+            Data for header keywords.
+        """
+        super().start_streaming_mode(exp_time, static_data)
+        with vimba.Vimba.get_instance():
```

Comment:
If I follow your code changes correctly, this context manager is started within a context manager for the same call to `get_instance`. Is there any way to reuse that instance?

Reply:
The library creator does not recommend reusing the context managers manually. Doing it this way follows their convention.